### PR TITLE
feat(EMS-1320): Completed application - Lock application

### PR DESCRIPTION
**Pull request by Zainzzkk**
# Changes made: 
* Added application status middleware for checking if application is completed
* Added template and controller for cannot-access-application-submitted
* Added e2e test
* Commented out test in submit-an-application as cannot access application after submission

### CODE REVIEW

File: src/ui/server/middleware/insurance/application-status/index.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  if (isApplicationSubmittedRoute(url)) {
+    return next();
+  }
+
+  /**
+   * If the status is submitted,
+   * dont allow the user to view/access the application and redirect to NO_ACCESS_APPLICATION_SUBMITTED.
+   * Otherwise, redirect to the application.
+   */
+  if (res.locals.application) {
```
Comment (abhi-markan) [inline src/ui/server/middleware/insurance/application-status/index.ts:22]: Do we need optional chaining here?

**Review by abhi-markan — COMMENTED** [review]
Minor comment to address please.

File: src/ui/server/controllers/insurance/no-access-application-submitted/index.test.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    it('should have the correct template defined', () => {
+      expect(TEMPLATE).toEqual(TEMPLATES.INSURANCE.NO_ACCESS_APPLICATION_SUBMITTED);
+    });
+  });
+
+  describe('get', () => {
+    it('should render template', () => {
+      get(req, res);
+
+      const expectedVariables = {
```
Comment (ttbarnes) [inline src/ui/server/controllers/insurance/no-access-application-submitted/index.test.ts:27]: Don't need the spread operator here can just be `expectedVariables = insuranceCorePageVariables({ ...`

File: src/ui/server/controllers/insurance/no-access-application-submitted/index.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { PAGES } from '../../../content-strings';
+import { TEMPLATES } from '../../../constants';
+import { Request, Response } from '../../../../types';
+import insuranceCorePageVariables from '../../../helpers/page-variables/core/insurance';
+
+export const TEMPLATE = TEMPLATES.INSURANCE.NO_ACCESS_APPLICATION_SUBMITTED;
+
+export const get = (req: Request, res: Response) =>
```
Comment (ttbarnes) [inline src/ui/server/controllers/insurance/no-access-application-submitted/index.ts:8]: documentation

File: src/ui/server/middleware/insurance/application-status/index.ts (hunk 1 above)
Comment (ttbarnes) [reply src/ui/server/middleware/insurance/application-status/index.ts:22]: The middleware run before this middleware will catch this, so before this is run, if there is no application in `res.locals` it'll redirect to `NO_ACCESS_TO_APPLICATION`

File: src/ui/server/middleware/insurance/application-status/index.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import { INSURANCE_ROUTES } from '../../../constants/routes/insurance';
+import { APPLICATION } from '../../../constants';
+import { Next, Request, Response } from '../../../../types';
+
+const { APPLICATION_SUBMITTED, NO_ACCESS_APPLICATION_SUBMITTED } = INSURANCE_ROUTES;
+
+export const applicationStatusMiddleware = async (req: Request, res: Response, next: Next) => {
```
Comment (ttbarnes) [inline src/ui/server/middleware/insurance/application-status/index.ts:7]: documentation

**Review by ttbarnes — DISMISSED** [review]

**Review by ttbarnes — APPROVED** [review]

**Pull request merged**